app/models/build/locking.rb

```
@@ -0,0 +1,47 @@
+module Build
+  module Locking
+    extend self
+
+    # Create an exclusive lock and yield
+    def lock(lock_name, options = {})
+      mutex = build_mutex(lock_name, options)
+
+      begin
+        mutex.lock
```

https://github.com/kenn/redis-mutex/blob/master/lib/redis/mutex.rb#L31 it will try to obtain the block for up to 240 seconds and then if it still can't obtain the lock it will fail silently. I will add an exception for failing to obtain a lock.